src/main/java/org/sonarlint/intellij/config/project/SonarLintProjectBindPanel.java

```diff
@@ -146,11 +147,12 @@ public String getSelectedProjectKey() {
   private void onConnectionSelected() {
     ApplicationManager.getApplication().assertIsDispatchThread();
 
-    var connectionSelected = getSelectedConnection() != null;
-    projectKeyTextField.setEnabled(connectionSelected);
-    projectKeyTextField.setEditable(connectionSelected);
-    searchProjectButton.setEnabled(connectionSelected);
-    updateStorageButton.setEnabled(connectionSelected);
+    var selectedConnection = getSelectedConnection();
+    var isAnyConnectionSelected = selectedConnection != null;
+    projectKeyTextField.setEnabled(isAnyConnectionSelected);
+    projectKeyTextField.setEditable(isAnyConnectionSelected);
+    searchProjectButton.setEnabled(isAnyConnectionSelected);
+    updateStorageButton.setEnabled(selectedConnection != null && getGlobalSettings().exists(selectedConnection.getName()));
```

Comment:
we can use isAnyConnectionSelected instead of selectedConnection != null here

Reply:
Good point, I was worried that the IDE would complain about selectedConnection being nullable but it's fine